src/Math/Geometry/Rectangle.cs

```diff
@@ -0,0 +1,162 @@
+
+        /// <summary>
+        /// The y-coordinate of the top boundary of the rectangle.
+        /// </summary>
+        public float Top
+        {
+            get { return this.y; }
+        }
+
+        /// <summary>
+        /// The y-coordinate of the bottom boundary of the rectangle.
+        /// </summary>
+        public float Bottom
+        {
+            get { return this.y + this.height; }
+        }
+
+        /// <summary>
+        /// The coordinates of the top left corner of the rectangle.
+        /// </summary>
+        public Vector2 TopLeft
+        {
+            get { return new Vector2(this.Left, this.Top); }
+        }
+
+        /// <summary>
+        /// The coordinates of the top right corner of the rectangle.
+        /// </summary>
+        public Vector2 TopRight
+        {
+            get { return new Vector2(this.Right, this.Top); }
+        }
+
+        /// <summary>
+        /// The coordinates of the bottom left corner of the rectangle.
+        /// </summary>
+        public Vector2 BottomLeft
+        {
+            get { return new Vector2(this.Left, this.Bottom); }
+        }
+
+        /// <summary>
+        /// The coordinates of the bottom right corner of the rectangle.
+        /// </summary>
+        public Vector2 BottomRight
+        {
+            get { return new Vector2(this.Right, this.Bottom); }
+        }
+        
+        /// <summary>
+        /// The coordinates of the center of mass of the rectangle.
+        /// </summary>
+        public Vector2 Center
+        {
+            get { return new Vector2(this.x + this.width * 0.5f, this.y + this.height * 0.5f); }
+        }
+        #endregion
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="x">X-coordinate of the upper left corner.</param>
+        /// <param name="y">Y-coordinate of the upper left corner.</param>
+        /// <param name="width">Width of the rectangle.</param>
+        /// <param name="height">Height of the rectangle.</param>
+        public Rectangle(float x, float y, float width, float height)
+        {
+            this.x = x;
+            this.y = y;
+            this.width = width;
+            this.height = height;
+        }
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="xy">Coordinates of the upper left corner.</param>
+        /// <param name="width">Width of the rectangle.</param>
+        /// <param name="height">Height of the rectangle.</param>
+        public Rectangle(Vector2 xy, float width, float height)
+            : this(xy.X, xy.Y, width, height) { }
+
+        /// <summary>
+        /// Creates a new rectangle.
+        /// </summary>
+        /// <param name="upperLeft">Coordinates of the upper left corner.</param>
+        /// <param name="bottomRight">Coordinates of the bottom right corner.</param>
+        public Rectangle(Vector2 upperLeft, Vector2 bottomRight)
```

Comment:
I have seen vectors used for sizes all over the place, after all a size is just s relative vector, while a point is an absolute one. Also, while the parameters are named nicely, one usually does not see their names, when reading code that uses them, i.e. `var r = new Rectangle(new Vector2(1, 1), new Vector2(2, 2));`

It is also not very consistent with the other constructors, which DO require you to specify the size, and not the coordinates of the bottom left corner. In fact, next to the other two constructors, I would assume this one wants a size, if I did not know better.